### ISSUE TITLE
SQL validator non reliable: refactoring required

### ISSUE THREAD

**Issue by tobiscr** [kyma-incubator/reconciler#937]
**Description**

The SQL validator in the mothership is non-reliable and allows also invalid SQL queries:

A refactoring is required which removes static-values in the SQL statement by placeholders instead search for non-placeholder values in a query.

AC:
* SQL validator replaces static values by placeholders in an SQL query
* Query-object is no longer injecting placeholders into SQL queries

**Expected result**

SQL queries are consistently using placeholders instead of static values.

**Actual result**

Some queries are using static values and are potentially risky as sql-injection would be possible.

**Steps to reproduce**

**Troubleshooting**

**Comment by jakobmoellersap** [comment]
relates to https://github.com/kyma-incubator/reconciler/issues/503

**Comment by jakobmoellersap** [comment]
When investigating into this I stumbled upon a few issues that I think are key to how we solve this: Originally I planned to only use positional arguments in our prepared statement like `$1`. However this quickly becomes an issue if the sql is not generated sequentially but independant of each-other (e.g. the inventory filters use this approach). This would mean that the individual sequel blocks may not know of each others arguments and would mistakenly assign a positional arg twice. 

The only way I know of to circumvent this issue is a named parameterized query, e.g. for Oracle DB `@some_variable`. However this works natively only if the driver supports it. 

Then one can make use of `sql.NamedArg`and be done with this. Bad news is that PostgreSQL is one of the few DBs that do NOT support this syntax but only positional (postgres only supports named arguments for functions/stored procedures). Usually this is circumvented by ORM or Driver by replacing the named variables with positional args on the fly, but since we are custom here we do not gain this by default.

I tried to manually operate the filters on named arguments but because postgres does not have native support for native parameterized queries, I had to try to manually replace them with positional arguments. I attempted this in https://github.com/jakobmoellersap/reconciler/tree/fix/sql-injection-937 but ultimately it leads to many edge cases that could still be vulnerable (e.g. mixing of positional and named args) and i do not feel comfortable with including without significant maintenance effort in tests.

As a result, I will try to combine this query model with the GORM query builder from #897 and then make use of its named argument prepared statement based on https://gorm.io/docs/sql_builder.html#Named-Argument to tackle it holistically through the ORM and plug the query into `ExecRaw` from our Library at first.
